appOPHD/States/MapViewState.h

```diff
@@ -118,6 +118,8 @@ class MapViewState : public Wrapper
 	QuitSignal::Source& quit() { return mQuitSignal; }
 	MapChangedSignal::Source& mapChanged() { return mMapChangedSignal; }
 
+	FileIo::FileSaveDelegate fileSaveDelegate() { return {this, &MapViewState::onSaveGame}; }
```

Comment:
This feels a bit weird. Maybe `onSaveGame` should be `public`. Effectively this is making the `onSaveGame` method `public`, so might as well make the original method `public` rather than wrapping access in a new `public` method.

With `public` access to `MapViewState::onSaveGame`, the `Delegate` could be constructed in the caller. Or alternatively, the caller could have it's own handler member, which calls `MapViewState::onSaveGame`. Or perhaps the caller's handler could make a direct call to `save`. Maybe `save` should be made `public`, and perhaps we don't really need `MapViewState::onSaveGame`, since that class will no longer be directly responsible for this event, since it's no longer in control of the `FileIo` dialog.